Sources/Session.swift

```diff
@@ -42,6 +47,22 @@ public class Session<C: Client> {
 
         return task
     }
+    
+    public func dataTask<C: Call>(for call: C) async throws -> Result<C.Parser.OutputType> {
+        return try await withCheckedThrowingContinuation({ (continuation: CheckedContinuation<Result<C.Parser.OutputType>, Error>) in
```

Comment:
We should be able to use the built in URLSession async functions instead of a continuation wrapping.

The other possibility would be to use the continuation but just call the existing `dataTask(for call:completion:)` function of the Session. This would have the benefit of not having duplicate code.